tracer/src/Datadog.Trace/Logging/Internal/Sinks/ConsoleSink.cs

```diff
@@ -0,0 +1,136 @@
+﻿// <copyright file="ConsoleSink.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Serilog.Core;
+using Datadog.Trace.Vendors.Serilog.Events;
+using Datadog.Trace.Vendors.Serilog.Formatting.Display;
+
+/// <summary>
+/// Loosely based on https://github.com/manigandham/serilog-sinks-fastconsole
+/// </summary>
+internal class ConsoleSink : ILogEventSink, IDisposable
+{
+    private readonly StringWriter _bufferWriter = new();
+    private readonly TextWriter _consoleWriter;
+    private readonly BoundedConcurrentQueue<LogEvent?> _writeQueue;
+    private readonly Task _writeTask;
+
+    private readonly MessageTemplateTextFormatter _textFormatter;
+    private readonly ManualResetEventSlim _serializationMutex = new(initialState: false);
+    private readonly TaskCompletionSource<bool> _processExit = new();
+
+    public ConsoleSink(string messageTemplate, int queueLimit, TextWriter? consoleWriter = null)
+    {
+        _consoleWriter = consoleWriter ?? new StreamWriter(Console.OpenStandardOutput(), Console.OutputEncoding) { AutoFlush = true };
```

Comment:
Can't you use `Console.Out` instead of `Console.StandardOutput()`? Otherwise I believe you lose the synchronization with messages written with `Console.Write`

Reply:
We specifically don't do that to avoid the global locking that tanks perf... (see comments on line 69-72)

Similarly, we buffer in-memory for the same reason, to minimise calls to `Write`